client/rpc/src/eth/execute.rs

```diff
@@ -713,10 +715,15 @@ pub fn error_on_execution_failure(reason: &ExitReason, data: &[u8]) -> Result<()
 			))
 		}
 		ExitReason::Revert(_) => {
+			const OFFSET_START: usize = 4;
 			const LEN_START: usize = 36;
 			const MESSAGE_START: usize = 68;
 
 			let mut message = "VM Exception while processing transaction: revert".to_string();
+			// If error has no selector
+			if data.len() < OFFSET_START {
+				return Err(crate::err(JSON_RPC_ERROR_DEFAULT, message, Some(data)));
+			}
```

Comment:
If there no selector, it should indeed not be decoded but it is still a Revert and thus should return a `JSON_RPC_ERROR_REVERT` error.

```suggestion
			// If error has no selector
			if data.len() < OFFSET_START {
				return Err(crate::err(JSON_RPC_ERROR_REVERT, message, Some(data)));
			}
```